src/vs/workbench/contrib/extensions/browser/extensionsActions.ts

```diff
@@ -859,14 +871,39 @@ export class UpdateAction extends ExtensionAction {
 		}
 	}
 
-	private getLabel(extension?: IExtension): string {
-		if (!extension?.outdated) {
-			return localize('updateAction', "Update");
+	protected getLabel(): string {
+		return localize('update', "Update");
+	}
+}
+
+export class SkipUpdateAction extends AbstractUpdateAction {
+
+	constructor(
+		@IExtensionsWorkbenchService override readonly extensionsWorkbenchService: IExtensionsWorkbenchService
+	) {
+		super(`extensions.ignoreUpdates`, '', extensionsWorkbenchService);
+	}
+
+	override update() {
+		super.update();
+		if (this.extension) {
+			this._checked = this.extensionsWorkbenchService.getExtensionIgnoresUpdate(this.extension);
 		}
-		if (extension.outdatedTargetPlatform) {
-			return localize('updateToTargetPlatformVersion', "Update to {0} version", TargetPlatformToString(extension.gallery!.properties.targetPlatform));
+	}
+
+	override async run(): Promise<any> {
+		if (!this.extension) {
+			return;
 		}
-		return localize('updateToLatestVersion', "Update to {0}", extension.latestVersion);
+		alert(localize('ignoreExtensionUpdate', "Ignoring {0} updates", this.extension.displayName));
+		const newIgnoresAutoUpdates = !this.extensionsWorkbenchService.getExtensionIgnoresUpdate(this.extension);
+		this.extensionsWorkbenchService.setExtensionIgnoresUpdate(this.extension, newIgnoresAutoUpdates);
+		this.update();
```

Comment:
Why do we need to trigger update here? I would assume `IExtensionsWorkbenchService` triggers a change event after the action that updates this action

Reply:
Without the update, the checkmark is not being updated all the time when pressed. I agree it's very weird as the `ActionWithDropDownAction` listens on the event and then updates all child actions.

Reply:
But `setExtensionIgnoresUpdate` is not triggering the `onChange` event? I think it shall do right?

Reply:
Yes, that works